hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/ZookeeperWorkloadConstraintsRepository.java

```diff
@@ -0,0 +1,57 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload.constraints;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.curator.framework.CuratorFramework;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.api.TopicName;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperBasedRepository;
+import pl.allegro.tech.hermes.infrastructure.zookeeper.ZookeeperPaths;
+
+import java.util.HashMap;
+
+import static org.apache.commons.lang.ArrayUtils.isNotEmpty;
+
+public class ZookeeperWorkloadConstraintsRepository extends ZookeeperBasedRepository implements WorkloadConstraintsRepository {
```

Comment:
I think we could create a `PathChildrenCache` here, the structure is flat and the data should not change often, so we could skip reading it on each algorithm iteration. What do you think?

Reply:
Done

Reply:
Done

Reply:
Take a look at `ZookeeperWorkloadConstraintsPathChildrenCache`.